Dockerfile

```diff
@@ -18,16 +18,19 @@ WORKDIR /opt/nodemcu-firmware
 # - FLOAT_ONLY=1, if you want the floating point firmware
 #
 # What the commands do:
-# - store the Git branch in 'BRANCH'
+# - store the Git branch in 'BRANCH' if IMAGE_NAME not set
 # - unpack esp-open-sdk.tar.gz in a directory that is NOT the bound mount directory (i.e. inside the Docker image)
 # - remove all files in <firmware-dir>/bin
 # - make a float build if !only-integer
 # - copy and rename the mapfile to bin/
 # - make an integer build
 # - copy and rename the mapfile to bin/
-CMD BRANCH="$(git rev-parse --abbrev-ref HEAD)" && \
-    BUILD_DATE="$(date +%Y%m%d-%H%M)" && \
-    if [ -z "$IMAGE_NAME" ]; then IMAGE_NAME=${BRANCH}_${BUILD_DATE}; else true; fi && \
+CMD BUILD_DATE="$(date +%Y%m%d-%H%M)" && \
+    if [ -z "$IMAGE_NAME" ]; then \
```

Comment:
The use case is when the nodemcu-firmware project that the docker build is run on is not checked out as a standalone project, but as a submodule of another git project. The way that git handles submodules is that, in the submodule, instead of having a `.git/` directory with metadata, there is a `.git` file containing a reference to the project's metadata that is stored as a subdirectory of the main project's metadata, like `gitdir: ../.git/modules/nodemcu-firmware`. When your script is now run inside the docker container, this link points to a directory that is not accessible, because it is outside of the docker container, and the build will fail because of this.
Using git only in the case where IMAGE_NAME is not explicitly set at least gives the user a way to work around this.